src/AggregateFunctions/AggregateFunctionNull.h

```diff
@@ -414,6 +414,109 @@ class AggregateFunctionNullVariadic final
         this->nested_function->add(this->nestedPlace(place), nested_columns, row_num, arena);
     }
 
+    void addBatchSinglePlace(
+        size_t row_begin,
+        size_t row_end,
+        AggregateDataPtr __restrict place,
+        const IColumn ** columns,
+        Arena * arena,
+        ssize_t if_argument_pos) const final
+    {
+        /// We are going to merge all the flags into a single one to be able to call the nested batching functions
+        std::vector<const UInt8 *> nullable_filters;
+        const IColumn * nested_columns[number_of_arguments];
+
+        std::unique_ptr<UInt8[]> final_flags = nullptr;
+        const UInt8 * final_flags_ptr = nullptr;
+
+        if (if_argument_pos >= 0)
+        {
+            final_flags = std::make_unique<UInt8[]>(row_end);
```

Comment:
why not create array with size of (row_end - row_begin)?

Reply:
Although it would save space it might cause problems in the called functions which expect a full array. They shouldn’t dereference the beginning of the array (it’d be a bug if they do), but passing a pointer to memory that can’t be derefenced (you’d need to pass ‘data() - row_begin’) is tricky and I’d only consider it if the improvement was noticeable.

Also, back then the usage of row_begin different than 0 was minimal. Probably close to 0. That changed recently